datalad_service/handlers/files.py

```diff
@@ -37,8 +38,9 @@ def on_get(self, req, resp, dataset, filename=None, snapshot='HEAD'):
 
                 # if fd is a reference to the git annex file location, try to open that file. otherwise,
                 # it is a file to be returned
-                if fd.find('.git/annex') >= 0:
-                    fd = fd[fd.find('.git/annex'):] # remove leading relative folder paths
+                if fd.startswith('.git/annex'):
```

Comment:
this needs to be how it was because nested files take a name of the form "../../.git/annex/filepath'. searching for the index guaranteed that the leading relative path references were ignored in the pathname